src/Microsoft.OData.Client/DataServiceActionQueryOfT.cs

```diff
@@ -35,9 +35,34 @@ public sealed class DataServiceActionQuery<T>
         /// <param name="parameters">Parameters of this action.</param>
         public DataServiceActionQuery(DataServiceContext context, string requestUriString, params BodyOperationParameter[] parameters)
         {
-            this.Context = context;
+            this.context = context;
             this.RequestUri = new Uri(requestUriString);
-            this.Parameters = parameters;
+            this.parameters = parameters;
+        }
+
+        /// <summary>
+        /// Gets the <see cref="DataServiceContext"/> that will be used to execute the action query
+        /// </summary>
+        public DataServiceContext Context
+        {
+            get
+            {
+                return this.context;
+            }
+        }
+
+        /// <summary>
+        /// Gets the <see cref="BodyOperationParameter"/>s that will be provided to the action when the query is executed
+        /// </summary>
+        public IEnumerable<BodyOperationParameter> Parameters
+        {
+            get
+            {
+                foreach (var element in this.parameters)
+                {
+                    yield return element;
+                }
+            }

```

Comment:
I assume this is done so that the user does not modify the array. But it seems excessive to me, especially given that we do not seem modify the list internally. I think by the time they're casting it back into array to try and modify it, they're already assuming any risks that come with that. But if we really want to prevent that, then maybe it's better to hand them a `ReadOnlyCollection` that's computed once so we don't have to generate the enumerable each time the property is accessed.